src/Illuminate/Database/DatabaseServiceProvider.php

```diff
@@ -45,6 +46,30 @@ public function register()
         $this->registerEloquentFactory();
 
         $this->registerQueueableEntityResolver();
+
+        $this->registerDBALTypes();
+    }
+
+    /**
+     * Add any custom types to the DBAL library.
+     *
+     * @return void
+     */
+    protected function registerDBALTypes()
+    {
+        if (! class_exists(Type::class)) {
+            return;
+        }
+
+        $types = config('database.dbal.types');
+
+        if (! is_array($types)) {
+            return;
+        }
```

Comment:
You can simplify this by providing a empty array as a default.

```suggestion
        $types = config('database.dbal.types', []);
```

Reply:
This change works for missing configurations but not for invalid configurations such as 
```
(database)['dbal']['types'] = 'string value';
```

My example is a long way from the norm, I know.  So if expecting valid configuration and only accounting for missing configuration is Laraval's modus operandi then I'll go ahead with your suggestion.

Reply:
Yep, we usually expect them to provide proper config values.